test/test-doctests.js

```diff
@@ -1,536 +1,586 @@
-var testFile = require("sdk/self").data.load("doctests.js");
-
-eval(testFile);
+var data = require("sdk/self").data;
+var tabs = require("sdk/tabs");
+
+const url = "https://developer.mozilla.org/en-US/";
+
+function runTests(assert, done, name, desc, url, tests) {
+  tabs.open({
+    url: url,
+    onReady: tab => {
+      var worker = tabs.activeTab.attach({
+        contentScriptFile: [data.url("doctests.js")],
+        contentScript: "var testObj = docTests[self.options.name];" +
```

Comment:
Could this be another contentScriptFile? I am not really a fan of this string notation.